File: exercise2/Statically.java
package exercise2;

public class Statically {
    private static String glue = "Epoxy";
    String jello = "Jello";

    // in order to call methods of the class I created main method:
    public static void main(String[] args) {

        // to call non static method I need to create instance of the class
        Statically s = new Statically();
        s.moveAndShake();

        // Calling static methods
        stubborn();
        explain();

    }

    // this is public method
    public void moveAndShake() {
        System.out.println("Never stagnate in life.  Be agile, change, and improve...");
    }

    // static method return type is void we are just printing only
    public static void stubborn() {
        Statically s2 = new Statically();
        s2.moveAndShake();
        System.out.println("Please don't make me change... I just don't like it...");
    }



    public static void explain(){
        stubborn();
        System.out.println(glue);
        System.out.println("Explanations: ");
        System.out.println("1. Well first I created main method inside a class Statically class because I know there must be main method in each project in order to run our code ");
        System.out.println("2. For non static method I created instane of class using default constructor and then I invoked the method using instance of class that I just created");
        System.out.println("3. For static methods, in order to invoke them, I did not have to create instate of class I just invoked directly from main method ");
    }
}

File: exercise2/statically.js
class Statically {
    //fix this code, so all of the instantiation and console.logs() work.  Try not to change the signature of the class, unless absolutely necessary.
    static #glue = "Epoxy";
    jello = "Jello";

    moveAndShake() {
        console.log("Never stagnate in life.  Be agile, change, and improve...");
    }

    static stubborn() {
        // this.moveAndShake(); // this is non static method, We cannot invoke it inside a Class itself
        console.log("Please don't make me change... I just don't like it...");
    }

    static explain() {
        this.stubborn();
        console.log(Statically.#glue); // glue is a static so It can be called only from the Class itself
		console.log("I wrote explanation in each line where functions or variable invoked write an explanation for how the keyword static behaves differently in javascript vs. java. Another way of accessing #glue is using getter method, but I still have to call Class to access it")
    }

		getGlue (){
			return Statically.#glue;
		}


}

//get the below code to run correctly without errors:  Make sure you explain what you did in the the explain() method, defined in the class.  Do NOT remove the 'static' keyword from the explain() method.

const stats = new Statically();

stats.moveAndShake();
Statically.stubborn(); // static method can be called by Class only not by Class instances
Statically.explain(); // static method can be called by Class only not by Class instances

console.log(stats.getGlue()); // #glue cannot be invoked from instance either we need to use
stats.getGlue() // In order to make getGlue to work, I returned Statically.#glue inside a getGlue because glue is a static and private variable
// console.log(stats.jello);

// Once you have finished getting statistically.js to run, refactor the code here in Java.
